### PR TITLE
Enable configuration to avoid successfully written Table Row propagation as part of WriteResult for StreamingInserts

### PR DESCRIPTION
**Pull request by prodriguezdefino**
Adding option to avoid successfully inserted TableRow propagation as part of the WriteResult object.

There seem to be an indication that by always propagating the successful results of the StreamingInsert operation a long running and throughput intensive pipeline can get strained and become less stable than one that does not propagate this results unconditionally. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by prodriguezdefino** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by prodriguezdefino** [comment]
Run Spotless PreCommit

**Comment by prodriguezdefino** [comment]
Run Java PreCommit

**Comment by prodriguezdefino** [comment]
R: @arunpandianp 
R: @reuvenlax

**Comment by aaltay** [comment]
R: @johnjcasey

**Comment by prodriguezdefino** [comment]
> Overall these changes look good, but this needs at least a few tests

Added few checks to existing test that validates misconfigurations of this properties. 

There were no existing test that validates expectations on positive configurations (from previous existing functionality).

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java [hunk 1]
```diff
@@ -151,7 +151,9 @@ public PCollection<TableDestination> getSuccessfulTableLoads() {
   public PCollection<TableRow> getSuccessfulInserts() {
     if (successfulInserts == null) {
       throw new IllegalStateException(
-          "Retrieving successful inserts is only supported for streaming inserts.");
```
[the remark this answers is not in the document]
Comment (prodriguezdefino) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java:154]: Not as is, but possibly we should share the `Write.Method` to this object so it knows the context and create a message like `Retrieving successful inserts is only supported for streaming inserts (Write.Method currently as FILE_LOADS).`
Would that help?

**Comment by prodriguezdefino** [comment]
Run Spotless Apply

**Comment by prodriguezdefino** [comment]
fixes #22416

**Comment by prodriguezdefino** [comment]
@ JohnZZGithub

**Comment by johnjcasey** [comment]
LGTM

**Comment by aaltay** [comment]
Merging: tests are passing and @johnjcasey approved.

**Pull request merged**